### PR TITLE
Analytics Hub: Create date range section

### LINKED ISSUE #8144: [Analytics Hub] Create Date range UI section with fixed date range selection
Depends on #8145 

## Summary

The Analytics Hub must have a Selected Date Range section bar fixed at the top of the view 

The Date range section bar should display the currently selected date range information alongside the previous period, that's always related to the selected range type. 

## Behavior

If the current date is 2022-11-17, and the `Year to date` option is currently selected, we should display:

`Year to date Jan 1 - Nov 17, 2022 compared to Jan 1 - Nov 17, 2021`

The date range section follows the respective My Store tab used as Entry point (capture from the WCAndroid implementation):

https://user-images.githubusercontent.com/5920403/202512643-fb2739a3-c73b-46b2-91a6-f6fb9a595d2c.mp4

When hitting the `See more` button, the following behavior should happen:

- Today tab as entry point -> Date range selection as `Today` compared to `Yesterday`
- Week tab as entry point -> Date range selection as `Week to date` compared to same period 7 days ago
- Month tab as entry point -> Date range selection as `Month to date` compared to same period 1 month ago
- Year tab as entry point -> Date range selection as `Year to date` compared to same period 1 year ago

This date range selection must dictate the information that will be requested for the stats cards inside the Analytics Hub.

## Designs

The designs for the date range section changed quite a bit from what's in figma, what's implemented in WCAndroid, and what was suggested by our designer's review afterward.

The section must be positioned like this:
Image

But the section itself should looks like this:
Image

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Partially fix issue #8144 by introducing the Date Range UI section to the Analytics Hub main view through an AnalyticsTimeRangeCard.

Screen Capture
==========

How to Test
==========
- Launch the app
- Tap the "See More" button from the main dashboard
- See the template hub view

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift
Comment (Ecarrion) [inline WooCommerce/Classes/Extensions/UIImage+Woo.swift:1119]: Could you please update the `IconTests` file too? :-)

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubView.swift
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubView.swift]: Nit: Missing indentation!

```suggestion
                        .background(Color(uiColor: .listForeground))
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift:5]: Could you add a small documentation header to this type? It's nice to be able to glance at the view details without looking at the code.

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift]: This works great but what do you think about this alternative?
```swift
Image(uiImage: .calendar)
    .padding()
    .background(
        Circle().foregroundColor(Color(.systemGray6))
    )
```

I think it has some benefits like
- The hierarchy is more clear. We easily now that we are adding a background for the view.
- We close the possibility for someone to add stuff in the main `HStack` by mistake.

What do you think?

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift]: nit: You can add the `String.localizedStringWithFormat...` as a function in the localization menu to keep the view layout code clear!

**Review by Ecarrion — APPROVED** [review]
Great work @ThomazFB! Just left some suggestions, mostly to improve the readability of the code!

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift]: That's an awesome suggestion. Thanks for letting me know about this option! 🙇  It's fixed here: https://github.com/woocommerce/woocommerce-ios/pull/8159/commits/066840a7f1c64099fc25d2d4980a1f9a613785ef

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubView.swift
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubView.swift]: Oh, Indeed. I went ahead and adjusted for the other card's declaration in the file too! Fixed here: https://github.com/woocommerce/woocommerce-ios/pull/8159/commits/25799eadbb7ad2ce445f0712b796043ed6473b06

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift
Comment (ThomazFB) [reply WooCommerce/Classes/Extensions/UIImage+Woo.swift:1119]: Done! https://github.com/woocommerce/woocommerce-ios/pull/8159/commits/f0c2a37a200814f69709d64de84b88d3ba4a8a6f

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift]: Makes perfect sense. I've adjusted here: https://github.com/woocommerce/woocommerce-ios/pull/8159/commits/6f8a7a13234aff0eb83ee7edad3147ba6d98de6f

Let me know if you think it makes sense this way.

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift:5]: Done! https://github.com/woocommerce/woocommerce-ios/pull/8159/commits/4244db06d05e57e89d6932e5d4437c19ac3e8871

**Comment by ThomazFB** [comment]
@Ecarrion, I've addressed all your suggestions! Thanks for this careful review 🙇 The PR is ready for a second round!

**Review by Ecarrion — APPROVED** [review]
> @Ecarrion, I've addressed all your suggestions! Thanks for this careful review 🙇 The PR is ready for a second round!

Thanks for listening to my suggestions! 

FYI since I had already approved you can feel free to merge it without asking for a second review!

**Pull request merged**
